qiskit_dynamics/models/lindblad_model.py

```diff
@@ -0,0 +1,616 @@
+        else:
+            return self._operator_collection.dissipator_operators[0].shape[-1]
+
+    @property
+    def signals(self) -> Tuple[SignalList]:
+        """Gets the Model's Signals.
+
+        Returns:
+            Tuple[] with 0th entry storing the Hamiltonian signals
+            and the 1st entry storing the dissipator signals.
+        """
+        return (self._hamiltonian_signals, self._dissipator_signals)
+
+    @signals.setter
+    def signals(self, new_signals: Tuple[List[Signal]]):
+        """Set signals.
+
+        Raises:
+            QiskitError: If signals incompatible with operator structure.
+        """
+        hamiltonian_signals, dissipator_signals = new_signals
+
+        # set Hamiltonian signals
+        if hamiltonian_signals is None:
+            self._hamiltonian_signals = None
+        elif hamiltonian_signals is not None and self.get_hamiltonian_operators() is None:
+            raise QiskitError("Hamiltonian signals must be None if hamiltonian_operators is None.")
+        else:
+            # if signals is a list, instantiate a SignalList
+            if isinstance(hamiltonian_signals, list):
+                hamiltonian_signals = SignalList(hamiltonian_signals)
+
+            # if it isn't a SignalList by now, raise an error
+            if not isinstance(hamiltonian_signals, SignalList):
+                raise QiskitError("Hamiltonian signals specified in unaccepted format.")
+
+            # verify signal length is same as operators
+            if len(hamiltonian_signals) != len(self.get_hamiltonian_operators()):
+                raise QiskitError(
+                    "Hamiltonian signals need to have the same length as Hamiltonian operators."
+                )
+
+            self._hamiltonian_signals = hamiltonian_signals
+
+        # set dissipator signals
+        if dissipator_signals is None:
+            self._dissipator_signals = None
+        elif dissipator_signals is not None and self.get_dissipator_operators() is None:
+            raise QiskitError("Dissipator signals must be None if dissipator_operators is None.")
+        else:
+            # if signals is a list, instantiate a SignalList
+            if isinstance(dissipator_signals, list):
+                dissipator_signals = SignalList(dissipator_signals)
+
+            # if it isn't a SignalList by now, raise an error
+            if not isinstance(dissipator_signals, SignalList):
+                raise QiskitError("Dissipator signals specified in unaccepted format.")
+
+            # verify signal length is same as operators
+            if len(dissipator_signals) != len(self.get_dissipator_operators()):
+                raise QiskitError(
+                    "Dissipator signals need to have the same length as dissipator operators."
+                )
+
+            self._dissipator_signals = dissipator_signals
+
+    def get_static_hamiltonian(self, in_frame_basis: Optional[bool] = False) -> Array:
```

Comment:
Since we have some methods that use setters, and others that use these set/get functions an idea to make them all use setters would be to add a `in_frame_basis` property with setter, so that in these classes you could have something like:

```python
@property
def in_frame_basis(self):
    return self._in_frame_basis

@in_frame_basis.setter
def in_frame_basis(self, value):
    self._in_frame_basis = value

@property
def static_hamiltonian(self):
    op = self._operator_collection.static_hamiltonian
    if self._in_frame_basis or self.rotating_frame is None:
        return op
    else:
        return self.rotating_frame.operator_out_of_frame_basis(op)
 
@static_hamiltonian.setter
def (self, hamiltonian):
     # etc
```

I'm not sure how often these methods are actually used by users (didn't see them anywhere in tutorial notebooks), but after typing above I saw that mostly they were used in internal codes where you need the argument so maybe this change would be inconvenient there unless you also did something like:

```python
def _get_static_operator(self, in_frame_basis):
 
@property
def static_operator(self):
   return self._get_static_operator(self._in_frame_basis)
# etc
```

Reply:
Yeah they are almost exclusively used internally. I really like this suggestion. I think the approach of just making the `_get`... methods private seems best. Will think about it but will probably proceed with this.